--- nhd2-exp.org/src/gui/imageinfo.cpp
+++ nhd2-exp/src/gui/imageinfo.cpp
@@ -156,16 +156,17 @@
 	//imagename = config.getString("imagename", "neutrino-HD2").c_str();
 	const char * homepage = "https://github.com/mohousch/neutrinohd2";
 	//homepage = config.getString("homepage", "http://gitorious.org/open-duckbox-project-sh4").c_str();
-	const char * docs = "https://github.com/mohousch/neutrinohd2";
+//	const char * docs = "https://github.com/mohousch/neutrinohd2";
 	//docs = config.getString("docs", "http://wiki.neutrino-hd.de").c_str();
-	const char * forum = "https://forum.mbremer.de/viewforum.php?f=86";
+//	const char * forum = "https://forum.mbremer.de/viewforum.php?f=86";
 	//forum = config.getString("forum", "http://gitorious.org/open-duckbox-project-sh4").c_str();
 	//const char * version = config.getString("version", "1202201602031021").c_str();
+	const char * creator = "Greder";
 	
 	//static CFlashVersionInfo versionInfo(version);
 	const char * releaseCycle = PACKAGE_VERSION;
 	//releaseCycle = versionInfo.getReleaseCycle();
-	const char * imageType = "Snapshot"; //FIXME:
+	const char * imageType = "ffmpeg-2.8.19"; //FIXME:
 	//imageType = versionInfo.getType();
 
 	// image name
@@ -182,7 +183,7 @@
 	ypos += iheight;
 	paintLine(xpos, font_info, _("Built date: "));
 	paintLine(xpos + x_offset, font_info, _(__DATE__));
-	paintLine(xpos + x_offset + x_offset, font_info, _(__TIME__));
+//	paintLine(xpos + x_offset + x_offset, font_info, _(__TIME__));
 
 	// git rev
 	//ypos += iheight;
@@ -194,58 +195,30 @@
 	paintLine(xpos, font_info, _("Type:"));
 	paintLine(xpos + x_offset, font_info, imageType);
 
-	// 
-	ypos += iheight;
-
-	// doko
+	// git
 	ypos += iheight;
-	paintLine(xpos, font_info, _("Docs:"));
-	paintLine(xpos + x_offset, font_info, docs);
+	paintLine(xpos, font_info, _("Git:"));
+	paintLine(xpos + x_offset, font_info, homepage);
 
-	// forum
+	// 
 	ypos += iheight;
-	paintLine(xpos, font_info, _("Forum:"));
-	paintLine(xpos + x_offset, font_info, forum);
 
-	// homepage
+	// creater
 	ypos += iheight;
-	paintLine(xpos, font_info, _("Home page:"));
-	paintLine(xpos + x_offset, font_info, homepage);
+	paintLine(xpos, font_info, _("Creator:"));
+	paintLine(xpos + x_offset, font_info, creator);
 
 	// license
 	ypos += 5*iheight;
-	paintLine(xpos, font_info, _("License:"));
-	paintLine(xpos + x_offset, font_small, "This program is free software; you can redistribute it and/or modify");
-
-	ypos += sheight;
-	paintLine(xpos + x_offset, font_small, "it under the terms of the GNU General Public License as published by");
-
-	ypos += sheight;
-	paintLine(xpos + x_offset, font_small, "the Free Software Foundation; either version 2 of the License, or");
-
-	ypos += sheight;
-	paintLine(xpos + x_offset, font_small, "(at your option) any later version.");
-
-	ypos += sheight;
-	paintLine(xpos + x_offset, font_small, "This program is distributed in the hope that it will be useful,");
+	paintLine(xpos, font_info, _("Info:"));
+	paintLine(xpos + x_offset, font_small, "This program is free software.");
 
 	ypos += sheight;
-	paintLine(xpos + x_offset, font_small, "but WITHOUT ANY WARRANTY; without even the implied warranty of");
+	paintLine(xpos + x_offset, font_small, "My git for creating the image of neutrinoHD2 for the Opticum9500");
 
 	ypos += sheight;
-	paintLine(xpos + x_offset, font_small, "MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.");
+	paintLine(xpos + x_offset, font_small, "https://github.com/Greder/buildsystem");
 
 	ypos += sheight;
-	paintLine(xpos + x_offset, font_small, "See the GNU General Public License for more details.");
-
-	ypos += sheight;
-	paintLine(xpos + x_offset, font_small, "You should have received a copy of the GNU General Public License");
-
-	ypos += sheight;
-	paintLine(xpos + x_offset, font_small, "along with this program; if not, write to the Free Software");
-
-	ypos += sheight;
-	paintLine(xpos + x_offset, font_small, "Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.");	
+	paintLine(xpos + x_offset, font_small, "Best regards, Greder ツ");
 }
-
-
